src/main/java/discodeit/service/basic/BasicChannelService.java

```diff
@@ -1,143 +1,154 @@
+                .toList());
+
+        // 모든 public 채널 추가
+        channelRepo.findAll()
+                .stream()
+                .filter(channel -> channel.getType().equals(ChannelType.PUBLIC))
+                .forEach(channelListAll::add);
+
+        return channelListAll.stream()
+                .map(this::toDto)
+                .toList();
     }
 
     @Override
-    public Channel update(String channelId, Channel updateChannel) {
-        Map<String, Channel> channelData = channelRepo.loadAll();
-        if (!channelData.containsKey(channelId)) {
-            throw new IllegalArgumentException("[error] 존재하지 않는 채널 ID입니다.");
-        }
-        Channel originChannel = channelData.get(channelId);
+    public Channel update(UUID channelId, UpdateChannelRequest updateChannelRequest) {
+        Channel channel = channelRepo.findById(channelId)
+                .orElseThrow(() -> new NoSuchElementException("[error] 존재하지 않는 채널 ID입니다."));
 
-        if (isChannelNameDuplicate(updateChannel.getChannelName())) {
+        if (isChannelNameDuplicate(updateChannelRequest.newName())) {
             throw new IllegalArgumentException("[error] 이미 존재하는 채널 이름입니다.");
         }
 
-        originChannel.updateChannelName(updateChannel.getChannelName());
-        channelRepo.save(originChannel);
-        return originChannel;
+        if (channel.getType().equals(ChannelType.PRIVATE)) {
+            throw new IllegalArgumentException("[error] PRIVATE 채널은 수정할 수 없습니다.");
+        }
+
+        channel.update(updateChannelRequest.newName(), updateChannelRequest.newDescription());
+        return channelRepo.save(channel);
     }
 
     @Override
-    public void deleteChannel(String channelId) {
-        Map<String, Channel> channelData = channelRepo.loadAll();
-        if (!channelData.containsKey(channelId)) {
-            throw new IllegalArgumentException("[error] 존재하지 않는 채널 ID입니다.");
-        }
-
-        channelData.remove(channelId);
-        System.out.println("[삭제 완료]");
-        messageService.deleteByChannel(channelData.get(channelId));
-        channelRepo.delete(channelRepo.loadById(channelId));
+    public void delete(UUID channelId) {
+       channelRepo.deleteById(channelId);
     }
 
     @Override
-    public void addUser(String channelId, User user) {
-        if (!channelRepo.loadAll().containsKey(channelId)) {
-            throw new IllegalArgumentException("[error] 존재하지 않는 채널 ID입니다.");
-        }
+    public void addParticipant(UUID channelId, UUID userId) {
+        Channel channel = channelRepo.findById(channelId)
+                .orElseThrow(() -> new NoSuchElementException("[error] 존재하지 않는 채널 ID입니다."));
+        User user = userRepo.findById(userId)
+                .orElseThrow(() -> new NoSuchElementException("[error] 존재하지 않는 User ID입니다."));
 
-        Channel channel = channelRepo.loadById(channelId);
-        if (isUserDuplicate(channel, user.getUserId())) {
+        if (isUserDuplicate(channel, user.getId())) {
```

Comment:
```suggestion
        if (channel.containsParticipant(user.getId())) {
```
엔티티 쪽에서 수행하면 어때요? 코드 응집도를 엔티티로 모을 수 있을거 같네요!